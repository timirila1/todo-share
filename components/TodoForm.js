import React from 'react';
import {
    View,
    StyleSheet,
} from 'react-native';
import PropTypes from 'prop-types';

import { LabeledInput } from './LabeledInput';
import { CustomButton } from './CustomButton';

export class TodoForm extends React.Component {
    constructor(props) {
        super(props);
        const { todo = {} } = this.props;
        this.state = {
            title: todo.title || '',
            desc: todo.desc || '',
        };
    }

    isUpdateMode = () => !!(this.props.todo && this.props.todo.id)

    render() {
        const { onCancel, onSend } = this.props;

        return (
            <View style={styles.container}>
                <View style={styles.fieldset}>
                    <LabeledInput
                        label='Title'
                        value={this.state.title}
                        onChangeText={title => this.setState({ title })}
                        maxLength={40}
                    />
                </View>
                {/* <View style={styles.fieldset}>
                    <LabeledInput
                        size="small"
                        label='Description'
                        multiline={true}
                        numberOfLines={4}
                        value={this.state.desc}
                        onChangeText={desc => this.setState({ desc })}
                    />
                </View> */}
                <View style={[styles.fieldset, styles.buttons]}>
                    <CustomButton
                        title={'Cancel'}
                        mod="light"
                        onPress={onCancel}
                    />
                    <CustomButton
                        disabled={!this.state.title}
                        title={this.isUpdateMode() ? 'Update' : 'Add Todo'}
                        onPress={() => onSend({ ...this.state })}
                    />
                </View>
            </View>
        );
    }
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        padding: 20,
        paddingLeft: 40,
    },
    fieldset: {
        marginBottom: 20,
    },
    buttons: {
        flex: 1,
        flexDirection: 'row',
        justifyContent: 'space-between',
    },
});

TodoForm.propTypes = {
    onCancel: PropTypes.func,
    onSend: PropTypes.func,
    todo: PropTypes.object,
};
