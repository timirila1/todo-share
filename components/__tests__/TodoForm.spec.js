import 'react-native';
import React from 'react';
import { TodoForm } from '../TodoForm';
import renderer from 'react-test-renderer';

it('Render TodoForm', async () => {
    const tree = renderer.create(
        <TodoForm />
    ).toJSON();
    expect(tree).toMatchSnapshot();
});
