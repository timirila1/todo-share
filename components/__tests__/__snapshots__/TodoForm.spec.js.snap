// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render TodoForm 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "padding": 20,
      "paddingLeft": 40,
    }
  }
>
  <View
    style={
      Object {
        "marginBottom": 20,
      }
    }
  >
    <View>
      <Text
        style={
          Object {
            "color": "rgba(0, 0, 0, 0.6)",
          }
        }
      >
        Title
      </Text>
      <TextInput
        allowFontScaling={true}
        label="Title"
        maxLength={40}
        onChangeText={[Function]}
        style={
          Object {
            "borderBottomColor": "#5e1c73",
            "borderBottomWidth": 1,
            "fontSize": 24,
            "paddingBottom": 10,
            "paddingRight": 10,
            "paddingTop": 10,
          }
        }
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
  </View>
  <View
    style={
      Object {
        "marginBottom": 20,
      }
    }
  >
    <View>
      <Text
        style={
          Object {
            "color": "rgba(0, 0, 0, 0.6)",
          }
        }
      >
        Description
      </Text>
      <TextInput
        allowFontScaling={true}
        label="Description"
        multiline={true}
        numberOfLines={4}
        onChangeText={[Function]}
        style={
          Object {
            "borderBottomColor": "#5e1c73",
            "borderBottomWidth": 1,
            "fontSize": 24,
            "paddingBottom": 10,
            "paddingRight": 10,
            "paddingTop": 10,
          }
        }
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
  </View>
  <View
    style={
      Object {
        "marginBottom": 20,
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#fa6c35",
          "borderRadius": 10,
          "opacity": 1,
          "padding": 10,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 24,
            "fontWeight": "300",
            "lineHeight": 36,
            "textAlign": "center",
          }
        }
      >
        Add Todo
      </Text>
    </View>
  </View>
</View>
`;
